generic_expression_patterns_modules/download_recount2_data.R

```diff
@@ -1,82 +1,77 @@
 ## Run this once to setup environment
 ## Used R 3.6.3
-#if (!requireNamespace("BiocManager", quietly = TRUE))
-#  install.packages("BiocManager")
 
-#BiocManager::install("recount")
+library('recount')
 
 ## About
 # recount2 contains over 70,000 uniformly processed human RNA-seq samples spanning
 # TCGA, SRA and GTEx
-# library('recount')
 
 set.seed(123)
 
-get_recount2_template_experiment <- function(project_id,
-local_dir){
-
+get_recount2_template_experiment <- function(project_id, local_dir) {
   # This function downloads the gene expression data associated
   # with the input project_id and saves it in the local_dir
 
+  # Reset working directory to `local_dir`
+  setwd(local_dir)
+
   ## Data
 
   # Get data associated with project ids
-  # Download the RangedSummarizedExperiment object.
-  # The RangedSummarizedExperiment object for the counts summarized 
-  # at the gene level using the Gencode v25 (GRCh38.p7, CHR) annotation as provided by Gencode.
+  # Download the RSE (RangedSummarizedExperiment) object.
+  # The RSE object for the counts summarized at the gene level using the
+  # Gencode v25 (GRCh38.p7, CHR) annotation as provided by Gencode.
   if (!file.exists(file.path(project_id, 'rse_gene.Rdata'))) {
     download_study(project_id)
-  } 
-  load(file.path(project_id, 'rse_gene.Rdata'), verbose = TRUE)
+  }
+  load(file.path(project_id, 'rse_gene.Rdata'), verbose=TRUE)
 
-  # Counts are raw read counts from the sequencing run
   # Counts are the number of reads that map that each gene
 
-  # Rename counts for storage
-  assign(paste0("rse_gene_", project_id), rse_gene)
-
   rse_gene_scaled <- scale_counts(rse_gene)
   rse_gene_counts <- assays(rse_gene_scaled)$counts
   data_counts <- t(rse_gene_counts)
 
   ## Save counts matrix to file
-  write.table(data_counts,
-            paste(local_dir,'/recount2_template_data.tsv',sep=""),
-            sep='\t',
-            row.names=TRUE,
-            col.names=TRUE)
-  
-  ## Delete it if you don't need it anymore
-  unlink(project_id, recursive = TRUE)
```

Comment:
I added this line to remove the intermediate downloaded files, is this not needed because you deleted the `assign` call?

Reply:
I wanted to keep the downloaded file (at least temporally) so that if something went wrong, we don't have to download it again. (It is easy to delete them when the whole data process is done anyway.)